### PR TITLE
Add missing C mappings

### PR DESCRIPTION
**Pull request by drobilla**
Adds mappings for some missing standard C and POSIX types.

This is still not comprehensive, but suppresses a lot of false positives and incorrect suggestions in typical code.

### CODE REVIEW

**Comment by kimgr** [comment]
Thank you! Unfortunately we keep these symbol mappings in two places -- can you add the same mappings to `libc_symbol_map` in `iwyu_include_picker.cc`?

**Comment by drobilla** [comment]
Ah, I didn't notice that.  Sure, will do.

**Comment by drobilla** [comment]
I confess to not yet knowing where these are actually used, and thus not having tested this, but I pushed some fixups that add the corresponding bindings and it builds and doesn't make any additional tests fail anyway...

**Comment by drd-ableton** [comment]
Yep, good catch https://github.com/include-what-you-use/include-what-you-use/pull/638/commits/6e22ccb4e92f12c68447050a680a493759ebeafd

File: iwyu_include_picker.cc [hunk 1]
```diff
@@ -96,6 +96,8 @@ const IncludeMapEntry libc_symbol_map[] = {
   { "blkcnt_t", kPrivate, "<sys/stat.h>", kPublic },
   { "blkcnt_t", kPrivate, "<sys/types.h>", kPublic },
   { "blksize_t", kPrivate, "<sys/stat.h>", kPublic },
+  { "clock_t", kPrivate, "<sys/types.h>", kPublic },
+  { "clock_t", kPrivate, "<sys/time.h>", kPublic },
```
[the remark this answers is not in the document]
Comment (drobilla) [reply iwyu_include_picker.cc]: Yep, good catch https://github.com/include-what-you-use/include-what-you-use/pull/638/commits/6e22ccb4e92f12c68447050a680a493759ebeafd

**Comment by kimgr** [comment]
Looks good to me, thanks!

**Pull request merged**